equinox/nn/_embedding.py

```diff
@@ -126,18 +126,22 @@ def __call__(...):
                 def process_heads(
                     query_heads: Float[Array, "seq_length num_heads qk_size"],
                     key_heads: Float[Array, "seq_length num_heads qk_size"],
-                    value_heads: Float[Array, "seq_length num_heads vo_size"]
+                    value_heads: Float[Array, "seq_length num_heads vo_size"],
+                    index: Int[Array, ""]
```

Comment:
I think this will fail, as we don't pass this extra argument inside MHA?

Reply:
Yep, you're right. I've added a more complete example for this.